pkg/ingester/client/cortex.proto

```diff
@@ -33,6 +33,7 @@ message WriteRequest {
   }
   SourceEnum Source = 2;
   repeated MetricMetadata metadata = 3 [(gogoproto.nullable) = true];
+  bool skip_label_name_validation = 4;
```

Comment:
I'm wondering if we should use `google.protobuf.Any hints` approach here. Strictly speaking this isn't supported by Ingester service, so it doesn't belong here. But Distributor implements the same method. (When we expose Distributor via gRPC, I expect it will reuse the same `Push` method, with `cortex.WriteRequest` and `cortex.WriteResponse`)

Reply:
We need to keep `WriteRequest` compatible with Prometheus one. If we use the field number `4`, it's very likely it will clash next time Prometheus remote write adds a new field. To fix this, we could use a very high field number (eg. `1000`).

Reply:
@pstibrany is the `google.protobuf.Any hints` approach like the `options` field in the rules proto (https://github.com/cortexproject/cortex/blob/d62514185ca7678f0e77ab1a1f93d291409fd530/pkg/ruler/rules/rules.proto#L29), making this field extensible in the case of future config options? Or am I missing the boat here?